Inline property documentation and SDK docs

orangemug opened the pull request:
I've added inline documentation instead of tooltips as outlined in #567. I've also added SDK support to the property documentation where available. 

Fixes #567 (partial fix for #427)

Demo <https://2014-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

<img width="368" alt="Screenshot 2020-01-23 at 08 34 15" src="https://user-images.githubusercontent.com/235915/72968907-5bb5ab80-3dbc-11ea-99c9-4b38b4636065.png">

pathmapper commented:
Oh nice, just saw this PR after leaving comments in #425 and #567.

https://github.com/maputnik/editor/issues/567#issuecomment-577598795 is still valid, there's a part of the description missing for:
https://docs.mapbox.com/mapbox-gl-js/style-spec/#layout-line-line-cap
https://docs.mapbox.com/mapbox-gl-js/style-spec/#layout-line-line-join

Any chance to add it?

pathmapper commented:
These missing parts are nested in the style spec, e.g.:
https://github.com/mapbox/mapbox-gl-js/blob/master/src/style-spec/reference/v8.json#L783

orangemug commented:
Good spot @pathmapper, added in latest commit.

Demo <https://2020-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

I've given it a quick test and it looks good, I'm a little short on time today though

orangemug commented:
New build with fixes https://2031-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html

pathmapper commented:
I really like the inline documentation, some findings:

- `Sources` Modal:
When adding a source, the tooltip doesn't change when the source type is changed:

![image](https://user-images.githubusercontent.com/20856381/73117949-9864dc80-3f4d-11ea-831c-bc222f269690.png)


- There a no tooltips for the `Export` modal like they are for the token fields in the `Style Settings` modal.:

![image](https://user-images.githubusercontent.com/20856381/73118021-69029f80-3f4e-11ea-8803-850f92ef37f2.png)

orangemug commented:
Thanks @pathmapper I've added the docs to the export modal and updated the 'type' in the sources modal 

With regards to the data source type, I've gone with a simple message instead of listing the possible values. As explained in a code comment (https://github.com/maputnik/editor/pull/607/files#diff-3648d61a30bd24d5ecf41c74cfb5fc05R170-R176) 

> Kind of a hack because the type changes, however maputnik has `1..n`
> options per type, for example 
>
>  - 'geojson' - 'GeoJSON (URL)' and 'GeoJSON (JSON)'
>  - 'raster' - 'Raster (TileJSON URL)' and 'Raster (XYZ URL)'
>
> So we just ignore the values entirely as they are self explanatory

New build <https://2035-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

pathmapper commented on src/components/modals/SettingsModal.jsx:
```diff
@@ -234,7 +244,7 @@ class SettingsModal extends React.Component {
       </InputBlock>
 
-      <InputBlock label={"Style Renderer"} doc={"Choose the default Maputnik renderer for this style."}>
+      <InputBlock label={"Style Renderer"} fieldSpec={{doc: "Choose the default Maputnik renderer for this style."}}>
```
Maybe move this also to `fieldSpecAdditional`?

pathmapper commented:
> So we just ignore the values entirely as they are self explanatory

Good solution!

Sometimes there are URLs in the docs included, do we want them to be functional or leave them like they are now?

e.g. https://docs.mapbox.com/mapbox-gl-js/style-spec/root/#sprite

![grafik](https://user-images.githubusercontent.com/20856381/73177025-2f789280-410e-11ea-86aa-0d4717c46bb9.png)

orangemug commented:
@pathmapper ah yeah forgot to mention that. So the docs are actually markdown, I think lets leave that for now. They are readable at the moment, I'll find a good small markdown parser (probably remark) and update in another PR.

pathmapper commented:
@orangemug sure, looks good to me now. One comment pending (https://github.com/maputnik/editor/pull/607#pullrequestreview-348644678), but this doesn't need to be addressed, just thought that you wanted to move all Maputnik docs there.

orangemug commented:
Thanks @pathmapper I've fixed the code review comment, merging!
